tradeexecutor/strategy/trading_strategy_universe.py

```diff
@@ -550,4 +582,111 @@ def load_all_data(client: Client, time_frame: TimeBucket, execution_context: Exe
             pairs=pairs,
             candles=candles,
             liquidity=liquidity,
-        )
\ No newline at end of file
+        )
+
+
+def load_pair_data_for_single_exchange(
+        client: Client,
+        execution_context: ExecutionContext,
+        time_bucket: TimeBucket,
+        chain_id: ChainId,
+        exchange_slug: str,
+        pair_tickers: Set[Tuple[str, str]],
+        liquidity=False,
+) -> Dataset:
+    """Load pair data for a single decentralised exchange.
+
+    If you are not trading the full trading universe,
+    this function does a much smaller dataset download than
+    :py:func:`load_all_data`.
+
+    - This function uses optimised JSONL loading
+      via :py:meth:`~tradingstrategy.client.Client.fetch_candles_by_pair_ids`.
+
+    - Backtest data is never reloaded.
+      Furthermore, the data is stored in :py:class:`Client`
+      disk cache for the subsequent notebook and backtest runs.
+
+    - Live trading purges old data fields and reloads data
+
+    :param client:
+        Trading Strategy client instance
+
+    :param time_bucket:
+        The candle time frame
+
+    :param chain_id:
+        Which blockchain hosts our exchange
+
+    :param exchange_slug:
+        Which exchange hosts our trading pairs
+
+    :param exchange_slug:
+        Which exchange hosts our trading pairs
+
+    :param pair_tickers:
+        List of trading pair tickers as base token quote token tuples.
+        E.g. `[('WBNB', 'BUSD'), ('Cake', 'BUSD')]`.
+
+    :param liquidity:
+        Set true to load liquidity data as well
+
+    :param execution_context:
+        Defines if we are live or backtesting
+    """
+
+    assert isinstance(client, Client)
+    assert isinstance(time_bucket, TimeBucket)
+    assert isinstance(execution_context, ExecutionContext)
+    assert isinstance(chain_id, ChainId)
+    assert isinstance(exchange_slug, str)
```

Comment:
We should really think about adding static type checking to our pipeline to avoid such type checks in several places...